src/lwc/rd2StatusMappingSettings/rd2StatusMappingSettings.js

```diff
@@ -0,0 +1,382 @@
+    */
+    handleButtonClick(event) {
+        this.clearMessage();
+
+        switch (event.target.label) {
+            case editButtonLabel:
+                this.handleEdit();
+                break;
+            case cancelButtonLabel:
+                this.handleCancel();
+                break;
+            case saveButtonLabel:
+                this.handleSave();
+                break;
+        }
+    }
+
+    /***
+    * @description Displays page in the edit mode
+    */
+    handleEdit() {
+        if (this.records) {
+            this.records
+                .filter(mapping => mapping.isReadOnly === false)
+                .forEach(mapping => {
+                    mapping.oldState = mapping.state;
+                });
+        }
+
+        this.isViewMode = false;
+    }
+
+    /***
+    * @description Cancels the edit and displays the page in the view mode
+    */
+    handleCancel() {
+        //reset values to the values as they were before the edit
+        if (this.records) {
+            this.records
+                .filter(mapping => mapping.isReadOnly === false)
+                .forEach(mapping => {
+                    mapping.state = isNull(mapping.oldState) ? stateUnmappedLabel : mapping.oldState;
+                    mapping.oldState = null;
+                });
+        }
+        this.isViewMode = true;
+    }
+
+    /**
+    * @description Indicates the save button should be disabled when at least one status is unmapped
+    */
+    get isSaveDisabled() {
+        let disabled = false;
+
+        if (this.records) {
+            disabled = this.records
+                .filter(mapping => mapping.state === stateUnmappedLabel)
+                .length > 0;
+        }
+
+        return disabled;
+    }
+
+    /***
+    * @description Saves mapping records
+    */
+    handleSave() {
+        this.isLoading = true;
+        try {
+            let jsonRecords = JSON.stringify(this.records);
+
+            saveMapping({ jsonMapping: jsonRecords })
+                .then((deploymentId) => {
+                    this.registerDeploymentId(deploymentId);
+                })
+                .catch((error) => {
+                    this.handleError(error);
+                });
+        } catch (error) {
+            this.handleError(error);
+        }
+    }
+
+    /***
+    * @description Registers deployment Id for the deployment monitoring and
+    * data and messages display based on its status
+    */
+    registerDeploymentId(deploymentId) {
+        this.showToast(deploymentInProgressMessage, toastVariant.INFO);
+
+        this._deploymentIds.add(deploymentId);
+
+        this.handleDeploymentProgress(deploymentId);
+    }
+
+    /***
+    * @description Starts polling for the deployment job progress details until the deployment completes
+    */
+    handleDeploymentProgress(deploymentId) {
+        var self = this;
+
+        this.deploymentTimer = setTimeout(function () {
```

Comment:
@RadSFDeveloper Have you looked at the `DeploymentEvent__e` usage for this rather than just plain polling? The BdiAdvancedMapping component uses this for their CMDT deployment. I think it might be more efficient than polling.

Reply:
Platform events do not work with Lightning Out. I did implement usage of `DeploymentEvent__e` but when the LWC was tested within VF context, it did not work. Good side of time spent is that I learned how to implement platform events. Unfortunately, it just does not work with Lightning Out.